thundra/integrations/aiohttp/client.py

```diff
@@ -0,0 +1,113 @@
+import aiohttp
+import logging
+
+from types import SimpleNamespace
+from thundra.opentracing.tracer import ThundraTracer
+from thundra.plugins.invocation import invocation_support
+from thundra import constants, config, utils
+
+logger = logging.getLogger(__name__)
+
+
+async def on_request_start(session, trace_config_ctx, params):
+    tracer = ThundraTracer.get_instance()
+    if not tracer.get_active_span():
+        return
+
+    url_dict = utils.parse_http_url(str(params.url), config.http_integration_url_path_depth())
+    scope = tracer.start_active_span(operation_name=url_dict.get('operation_name'), finish_on_close=False)
+    scope.span.class_name = constants.ClassNames['HTTP']
+    scope.span.domain_name = constants.DomainNames['API']
+    
+    tags = {
+            constants.SpanTags['OPERATION_TYPE']: params.method,
+            constants.HttpTags['HTTP_METHOD']: params.method,
+            constants.HttpTags['HTTP_URL']: url_dict.get('url'),
+            constants.HttpTags['HTTP_PATH']: url_dict.get('path'),
+            constants.HttpTags['HTTP_HOST']: url_dict.get('host'),
+            constants.HttpTags['QUERY_PARAMS']: url_dict.get('query'),
+            constants.SpanTags['TRIGGER_OPERATION_NAMES']: [invocation_support.function_name],
+            constants.SpanTags['TRIGGER_DOMAIN_NAME']: constants.LAMBDA_APPLICATION_DOMAIN_NAME,
+            constants.SpanTags['TRIGGER_CLASS_NAME']: constants.LAMBDA_APPLICATION_CLASS_NAME,
+            constants.SpanTags['TOPOLOGY_VERTEX']: True,
+        }
+
+    scope.span.tags = tags
+    try:
+        scope.span.on_started()
+        try:
+            params.headers.update({'x-thundra-span-id': scope.span.span_id})
+            scope.span.set_tag(constants.SpanTags['TRACE_LINKS'], [scope.span.span_id])
+        except Exception as e:
+            pass
+        trace_config_ctx.scope = scope
+    except Exception as e:
+        logger.error(e)
+
+
+async def on_request_chunk_sent(session, trace_config_ctx, params):
+    if not hasattr(trace_config_ctx, "scope"):
+        return
+    scope = trace_config_ctx.scope
+    if not config.http_body_masked():
+        body = params.chunk if params.chunk else ""
```

Comment:
when there are multiple chunks, last chunk will override the previous ones, right?

Reply:
yes, I wasn't sure if collecting all the chunks and sending them would be better

Reply:
not a big problem but I think for such cases, setting only the first chunk as tag might be better